Syscollector IT `test_syscollector_collectors_disabled` fails after #36234 moved STARTUP_MSG before DataClean

## Summary

Three parametrized cases of `test_syscollector_collectors_disabled` (`packages_disabled`, `packages_processes_disabled`, `network_disabled`) fail on every Linux and Windows IT run since #36234 (*Review agent info logs*) landed on `main` on 2026-05-25:

```
test_syscollector/test_configuration/test_syscollector_configuration.py:572: in test_syscollector_collectors_disabled
    assert log_monitor.callback_result, "Scan should start for enabled collectors"
E   AssertionError: Scan should start for enabled collectors
E   assert None
```

## Why

#36234 (commit `c2373cd450`, *feat: Change startup logs to minfo*) moved syscollector's startup INFO log from `Syscollector::syncLoop()` (post-DataClean) to `wm_sys_main()` (pre-DataClean), and updated `CB_MODULE_STARTED` in `qa-integration-framework` from `INFO: Module started.` to `INFO: Started \(pid: \d+\).` to match.

Before:

```
[INFO] Module started.        <-- syncLoop, AFTER handleNotifyDataClean
[INFO] Starting evaluation.   <-- Syscollector::scan()
```

After:

```
[INFO] Started (pid: N).      <-- wm_sys_main, BEFORE syscollector_start_ptr()
...
  handleNotifyDataClean() retries (m_dataCleanRetries * 60s; default 3 -> 120s wait)
...
[INFO] Starting evaluation.
```

The test follows `CB_MODULE_STARTED` with `log_monitor.start(callback=...CB_SCAN_STARTED, timeout=10)` — only 10 s. In the IT environment `remoted_simulator` does not ack DataClean, so every `notifyDisableCollectorsDataClean()` fails and `handleNotifyDataClean()` burns its full retry budget before `syncLoop()` -> `scan()` -> `"Starting evaluation."` is finally emitted. The 10 s window is no longer enough.

The `all_collectors_disabled` case is not affected — it `return`s early after the DataClean log and never reaches `CB_SCAN_STARTED`.

## Why #36234 didn't catch it

The syscollector IT path triggers in `.github/test_modules_linux.json` and `.github/test-modules-windows.json` listed `src/config/src/wmodules-syscollector.c` and `src/wazuh_modules/syscollector/**`, but **not** `src/wazuh_modules/src/wm_syscollector.c`. #36234's change lives in `wm_syscollector.c`, so the matrix builder didn't select the syscollector shard.

(Latent: the same matrix builder evaluates patterns with `PurePosixPath.match()`, which doesn't implement recursive `**` semantics — `src/wazuh_modules/syscollector/**` only matches files exactly one segment deep. Out of scope here, tracked separately if needed.)

## Reproducible CI evidence

- Failing run (on `main`, propagated to PR #36244): https://github.com/wazuh/wazuh/actions/runs/26451549181/job/77962359127 — `3 failed, 34 passed, 2 skipped in 462.34s`
- Same failures on Windows: https://github.com/wazuh/wazuh/actions/runs/26451549182/job/77962390649 — `3 failed, 36 passed in 385.70s`

## Fix

PR #36446 — bump the `CB_SCAN_STARTED` timeout to 240 s and add `wm_syscollector.c` to the syscollector IT trigger paths so future changes to the wm-side entry point run the IT.

Verified green on Linux + Windows syscollector ITs on that PR.

cc @Antoniogm03 @jotacarma90

--- a/.github/test-modules-windows.json
+++ b/.github/test-modules-windows.json
@@ -141,6 +141,7 @@
         ".github/workflows/5_testintegration_windows-integration-tests.yml",
         ".github/test-modules-windows.json",
         "src/config/src/wmodules-syscollector.c",
+        "src/wazuh_modules/src/wm_syscollector.c",
         "src/wazuh_modules/syscollector/**",
         "src/win32/**",
         "src/Makefile",

--- a/.github/test_modules_linux.json
+++ b/.github/test_modules_linux.json
@@ -304,6 +304,7 @@
         ".github/workflows/5_testintegration_linux-integration-tests.yml",
         ".github/test_modules_linux.json",
         "src/config/src/wmodules-syscollector.c",
+        "src/wazuh_modules/src/wm_syscollector.c",
         "src/wazuh_modules/syscollector/**",
         "src/Makefile",
         "tests/integration/conftest.py",

--- a/tests/integration/test_syscollector/test_configuration/test_syscollector_configuration.py
+++ b/tests/integration/test_syscollector/test_configuration/test_syscollector_configuration.py
@@ -567,8 +567,12 @@ def test_syscollector_collectors_disabled(test_configuration, test_metadata, set
     log_monitor.start(callback=callbacks.generate_callback(patterns.CB_MODULE_STARTED), timeout=180)
     assert log_monitor.callback_result, "Module should continue running after DataClean"
 
-    # Check general scan has started (enabled collectors should scan)
-    log_monitor.start(callback=callbacks.generate_callback(patterns.CB_SCAN_STARTED), timeout=10)
+    # Check general scan has started (enabled collectors should scan).
+    # CB_MODULE_STARTED now fires in wm_sys_main *before* Syscollector::start()
+    # runs, so the gap until the first "Starting evaluation." can include the
+    # full handleNotifyDataClean retry budget (m_dataCleanRetries * 60s) when
+    # the test env's remoted_simulator does not ack DataClean.
+    log_monitor.start(callback=callbacks.generate_callback(patterns.CB_SCAN_STARTED), timeout=240)
     assert log_monitor.callback_result, "Scan should start for enabled collectors"
 
     # Map collector names to their scan callbacks

